x-pack/plugins/enterprise_search/server/lib/connectors/generate_connector_name.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { v4 as uuidv4 } from 'uuid';
+
+import { IScopedClusterClient } from '@kbn/core-elasticsearch-server';
+
+import { ErrorCode } from '../../../common/types/error_codes';
+
+import { toAlphanumeric } from '../../../common/utils/to_alphanumeric';
+import { indexOrAliasExists } from '../indices/exists_index';
+
+export const generateConnectorName = async (
+  client: IScopedClusterClient,
+  connectorType: string
+): Promise<{ apiKeyName: string; connectorName: string; indexName: string }> => {
+  const prefix = toAlphanumeric(connectorType);
+  if (!prefix || prefix.length === 0) {
+    throw new Error('Connector type is required');
```

Comment:
If we're defaulting to 'custom' in the frontend anyway, why not also default to 'custom' here?